### PR TITLE
[T4A4][F09-B1] Phua Kian Ming

### PR DESCRIPTION
**Pull request by daredevil999**
Applied Interface Segregation Principle by creating a Printable interface to extract printable form of details for each Person.

### CODE REVIEW

**Comment by daredevil999** [comment]
Ready for review

File: src/seedu/addressbook/data/person/Printable.java [hunk 1]
```diff
@@ -0,0 +1,12 @@
...
+
+/**
+ * An interface for concatenating a Person's details in the addressbook.
+ * Implementations should guarantee: details are present and not null, field values are validated.
+ */
+
+public interface Printable {
+    
+    public String getPrintableString();
+    public boolean isPrivate();
```
Comment (jeffryhartanto) [inline src/seedu/addressbook/data/person/Printable.java:11]: I don't think you need to make `isPrivate` method part of this interface, you can take a look at #1113 for reference.

File: src/seedu/addressbook/data/person/ReadOnlyPerson.java [hunk 2]
```diff
@@ -37,49 +37,56 @@ default boolean isSameStateAs(ReadOnlyPerson other) {
...
+       final String detailIsPrivate = "(private) ";
+       for (Printable p : printables) {
+           if (p.isPrivate()) {
+               builder.append(detailIsPrivate);
+           }
+           builder.append(p.getPrintableString())
+                   .append(" ");
+       }
+       return builder.toString();
+   }
```
Comment (jeffryhartanto) [inline src/seedu/addressbook/data/person/ReadOnlyPerson.java]: Can also consider putting this method in TextUI class.

**Comment by jeffryhartanto** [comment]
Some comments added. Please close the PR after reading comments.

**Pull request closed without merging**